frontend/game-rating-platform/src/compoments/GameTable.tsx

```diff
@@ -23,21 +16,42 @@ export function GameTable({
   // cache original game data for filter operations
   const [origGameData, setOrigGameData] = useState<GameProps[]>([]);
   const [gameData, setGameData] = useState<GameProps[]>([]);
+  const [activeModal, setActiveModal] = useState<boolean>(false);
+  const [clickedGame, setClickedGame] = useState<GameProps>({
+    id: 0,
+    title: "",
+    releaseDate: "",
+    developers: [],
+    description: "",
+    trailer: "",
+    avgRating: 0,
+    platform: "",
+  });
+
+  function openModal(e: SyntheticEvent<HTMLTableRowElement, Event>) {
+    setActiveModal(true);
+    let filteredData = gameData.filter(
+      (prop) => prop.title === e.currentTarget.id
+    );
+    console.log(filteredData);
+    if (filteredData.length !== 0) {
+      setClickedGame(filteredData[0]);
+    }
+  }
+
+  function closeModal() {
+    setActiveModal(false);
+  }
 
   useEffect(() => {
     const fetchGames = async (requestUrl: string) => {
-      const response = await fetch(requestUrl, {
-        method: "GET",
-        headers: {
-          "Content-Type": "application/json",
-        },
+      let data: GameProps[] = await fetchGameData(requestUrl);
```

Comment:
Du könntest noch über Fehlerbehandlung nachdenken, also das hier beispielsweise in einen try-catch-Block wrappen oder eine zentralere Fehlerbehandlung in gameData.tsx.